modules/openapi-generator/src/main/resources/typescript-axios/common.mustache

```diff
@@ -75,24 +75,34 @@ export const setOAuthToObject = async function (object: any, name: string, scope
     }
 }
 
+function setFlattenedQueryParams(urlSearchParams: Record<string, any>, parameter: any, key: string=""): void {
+    if (typeof parameter === "object") {
+        if (Array.isArray(parameter)) {
+            (parameter as any[]).forEach(item => setFlattenedQueryParams(urlSearchParams, item, key));
+        } 
+        else {
+            Object.keys(parameter as Record<string, any>).forEach(currentKey => 
+                setFlattenedQueryParams(urlSearchParams, parameter[currentKey], currentKey)
```

Comment:
```suggestion
                setFlattenedQueryParams(urlSearchParams, parameter[currentKey], `${key}.${currentKey}`)
```
since this never worked correctly with nested objects anyway (sorry for my previous comment to keep backwards compatibility, which was for a different typescript generator)

at a later stage, different options for key concatenation variants can be introduced, similar to 

https://github.com/OpenAPITools/openapi-generator/blob/b755ae288afe9b4d23de14e72ff6dc7735686690/modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache#L133